File: clients/python/moostodon_service.py
import asyncio

from typing import List
from kiota_http.httpx_request_adapter import HttpxRequestAdapter
from kiota_serialization_json.json_parse_node_factory import JsonParseNodeFactory
from kiota_serialization_json.json_serialization_writer_factory import JsonSerializationWriterFactory
from kiota_abstractions.authentication.anonymous_authentication_provider import AnonymousAuthenticationProvider
from sdk.models.status import Status
from sdk.models.account import Account
from sdk.moostodon_client import MoostodonClient



class MoostodonService():
    """
    This class is responsible for the communication with the Mastodon API using the MoostodonClient.
    """
    def __init__(self, base_url: str) -> None:
        """
        Instantiates a new MoostodonService and sets the default values.
        """
        provider = AnonymousAuthenticationProvider()
        request_adapter = HttpxRequestAdapter(provider)
        self.client = MoostodonClient(request_adapter)
        request_adapter.base_url = base_url
    
    def get_public_statuses(self) -> List[Status]:
        """
        Gets the statuses from the Moostodon API.
        Returns:
            The list of statuses.
        """
        statuses = asyncio.run(self.client.api().v1().timelines().public().get())
        return statuses

    def get_account(self, account_id: str) -> Account:
        """
        Get account details for a user.
        """
        
        return asyncio.run(self.client.api().v1().accounts_by_id(account_id).get())

File: clients/node/MoostodonService.ts
import { MoostodonClient } from "./sdk/moostodonClient";
import { FetchRequestAdapter } from '@microsoft/kiota-http-fetchlibrary';
import { AuthenticationProvider, RequestInformation } from "@microsoft/kiota-abstractions";
import { Account } from "./sdk/models/account";
import { Status } from "./sdk/models/status";

// Create Wrapper service around the MoostodonClient
export class MoostodonService {
    private _client: MoostodonClient;

    constructor(baseurl: string) {
        const adapter = new FetchRequestAdapter(new AnonymousAuthProvider());        
        this._client = new MoostodonClient(adapter);
        adapter.baseUrl = baseurl;
        console.log("initialized client with baseurl: " + adapter.baseUrl);
    }

    public get client(): MoostodonClient {
        return this._client;
    }

    // Get a  user's account details from the Moostodon Client
    public async getAccount(accountId: string): Promise<Account| undefined> {
        console.log("getting account");
        return await this._client.api.v1.accountsById(accountId).get();
    }

    // Get the mastodon public timeline
    public async getPublicTimeline(): Promise<Status[] | undefined> {
        return await this._client.api.v1.timelines.public.get();
    }
}

class AnonymousAuthProvider implements AuthenticationProvider {
    async authenticateRequest(request: RequestInformation, additionalAuthenticationContext?: Record<string, unknown>): Promise<void> {
        // do nothing
    }
}
